tests/foreman/api/test_puppetmodule.py

```diff
@@ -14,34 +14,24 @@
 
 :Upstream: No
 """
+import pytest
 from nailgun import entities
 
 from robottelo.constants import PUPPET_MODULE_NTP_PUPPETLABS
-from robottelo.decorators import tier1
-from robottelo.decorators import upgrade
 from robottelo.helpers import get_data_file
-from robottelo.test import APITestCase
 
 
-class RepositorySearchTestCase(APITestCase):
+class TestRepositorySearch:
     """Tests that search for puppet modules and filter by repository."""
 
-    @classmethod
-    def setUpClass(cls):
-        """Create a product. Make it available as ``cls.product``."""
-        super().setUpClass()
-        cls.product = entities.Product().create()
+    @pytest.fixture(scope='class')
+    def make_repo(self):
```

Comment:
In general, pytest prefers more shallow test structure, where tests are methods inside module. Tests can be grouped in classes, if needed, but usually that shouldn't be required. There's definitely no reason for fixture to be method of class. Please move it out of class, to be simple function in module.